docs/proposals/scheduling/20221227-cpu-static-reservation.md

```diff
@@ -0,0 +1,160 @@
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals/Future Work](#non-goalsfuture-work)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+      - [Story 1](#story-1)
+      - [Story 2](#story-2)
+    - [Implementation Details](#implementation-details)
+      - [koordlet](#koordlet)
+      - [koord-manager](#koord-manager)
+      - [koord-scheduler](#koord-scheduler)
+      - [example](#example)
+  - [Implementation History](#implementation-history)
+<!--te-->
+
+## Summary
+The proposal provides a scheduling mechanism to statically reserve resources on nodes, such as reserving 0-6 cores of cpu on the node.
+
+## Motivation
+
+
+### Goals
+
+- statically reserving cpu resources on node
+
+### Non-Goals/Future Work
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+As a k8s cluster administrator, there are not only containers running on the node of the k8s cluster, there may be other processes as well, in order to ensure the isolation of the containers and the original processes.
+At the cpu level, we usually want normal processes and containers to run on different cpu cores to reduce/avoid the impact of their resource usage.   
+eg：   
+totoal cpu on node:24;   
+expected performance:   
+normal process use:0-11;    
+container use:12-23 
+
+#### Story 2
+
+
+### Implementation Details
+
+#### koordlet   
+1. When koordlet report the `NodeResourceTopology` of the node, it will ignores the reserved cpu cores on node annotation. and will combine the reserved cpus of kubelet and `node.annotations`.   
+eg:  
+kubelet.args["reserved-cpus"]="6-7";  
+node.annotation["koordinator.sh/cpus-reserved"]="0-6"  
+then we hopes that tototal reserved cpus is "0-7",and it will be stored in `NodeResourceTopology` crd's annotation.
+
+2. cpu supress:  
+it will evict some pod whose qos is not lse. we should ignore the cpus reserved by node when supress with cpuset policy.  
+
+```
+cpuset(lsr) = processor(lsr) - cpusReservedByNode
+cpuset(!lse/lser) = processor(!lse/lser) - cpusReservedByNode
+``` 
+
+3. runtimehooks:  
+it will ignored the cpus reserved by `node.annotations` when create container. 
+```
+cpus(allocatable) = cpus(shared-pools) - cpus(reservedByNode)
+```
+
+#### koord-manager  
+should update batch-resource and origin resourse info on node, just like `memory`,`cpu`,`batch-memory`,`batch-cpu`.the implementation of this part depends on the `nodeResource`  controller in koord-manager:
+```
+Node(BE).Alloc = Node.Total - Node.Reserved - node.anno.reserved - System.Used - Pod(LS).Used
+Node.Alloc = Node.Alloc - node.anno.reserved
+```
+
+#### koord-scheduler
+
+pods in different qos:
+- for lse/lsr pod
+  - when scheduling
+    - if need core binding  
+      `nodeNumareResource` scheduler plugin can handle this situation, numa topo info of node will be considered when scheduling. 
+    - if don't need core binding  
+      `resourceFit` scheduler plugin handle it will be fine.
```

Comment:
We should implement new plugin to support the new reservation mechanism.  Do not modify or extend The k8s native plugin `NodeResourceFit`